### ISSUE TITLE
homebrew: Forumla Class Name doesn't always match what brew expects

### ISSUE THREAD

**Issue by pquerna** [goreleaser/goreleaser#5191]
### What happened?

If you have a app named `baton-1password`, goreleaser generates a class name for the homebrew Formula of `Baton1Password`, while homebrew expects a class name of `Baton1password` (lowercase p).

When in this situation, homebrew will error out like this:
```
In formula file: /opt/homebrew/Library/Taps/conductorone/homebrew-baton/Formula/baton-1password.rb
Expected to find class Baton1password, but only found: Baton1Password.
```

### How can we reproduce this?

https://github.com/ConductorOne/baton-1password/blob/main/.goreleaser.yaml#L68-L77

outputs:
https://github.com/ConductorOne/homebrew-baton/blob/main/Formula/baton-1password.rb

### goreleaser version

```bash
goreleaser/goreleaser-action@v6
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

This is a related issue to https://github.com/goreleaser/goreleaser/pull/3117 and https://github.com/goreleaser/goreleaser/issues/3116

Basically the improved function doesn't match all of the behaviors of the ruby function.

**Comment by caarlos0** [comment]
thanks for the issue and for the PR 🙏🏻
